OTHER-343: Port to Python 3

Pac23 opened the pull request:
Port to Python 3,after looking at the code not many changes occur to me.I have fixed what i have noticed,do let me know if i have missed anything.

https://tickets.metabrainz.org/browse/OTHER-343

samj1912 commented:
These are auto generated btw. Not sure if these should be manually edited. The next time we update the models these specific changes would be lost. I would suggest regenerating it using python3.

samj1912 commented:
I would update https://github.com/metabrainz/mb-rngpy/blob/master/update.sh

samj1912 commented:
https://github.com/metabrainz/mb-rngpy/blob/master/Makefile

Pac23 commented:
> I would update https://github.com/metabrainz/mb-rngpy/blob/master/update.sh

i updated the file to build using 3.5 ,get a error `cant find remote ref production error`

samj1912 commented:
You also need to update the makefile which is more important

Pac23 commented:
> You also need to update the makefile which is more important

done

samj1912 commented:
And... Try running them as well to make sure they work. You might have to update the pinned dependencies on genersteDS

samj1912 commented:
Also, just a FYI for @yvanzo and others. Once this PR is done, do not merge it to master. Please create a py3 branch and merge it. Once all of metabrainz/sir has been ported to py3, you can move this to master.

yvanzo commented:
@samj1912, thank you, I just created the branch `py3` out of `master` and will change the target of this PR if necessary before merging once it has been tested and approved.

Pac23 commented:
> And... Try running them as well to make sure they work. You might have to update the pinned dependencies on generateDS

updated the models using make

yvanzo commented on setup.py:
```diff
@@ -119,7 +118,7 @@ def download_url(version):
 
 setup(name="mb-rngpy",
-      python_requires="~=2.7",
+      python_requires="~=3.5",
```
Why is this particular version of Python 3.x required? (Documenting in commit message would be nice.)

Pac23 replied:
updated

yvanzo replied:
doesn’t seem so

Pac23 replied:
check commit message.generic py version

yvanzo commented on setup.py:
```diff
@@ -137,7 +136,10 @@ def download_url(version):
+                   "Programming Language :: Python :: 3",
+                   "Programming Language :: Python :: 3.5",
+                   "Programming Language :: Python :: 3.6",
+                   "Programming Language :: Python :: 3.7",
```
Same as above: Why these versions? (I have no idea, but it should be explained.)

Pac23 replied:
changed to classifier to a non specific one so any py3 version works with pypi.

yvanzo commented on Makefile:
```diff
@@ -3,7 +3,7 @@
 
 mbrng/__init__.py:
-	python setup.py version
+	python3 setup.py version
```
You’re right: this occurrence has to be the same as the first line in `setup.py`.

Pac23 commented on update.sh:
```diff
@@ -24,26 +24,16 @@ then
-    python_command=python2
-  elif type python &>/dev/null
-  then
-    python_command=python
```
since we have the virutal env being crated from upadate.sh which specifies the venv being py3 even the python shebang line from mbrng/ would default to using that itself

yvanzo replied:
Alright but then the error message error below should refer to `python3` instead of `python`.

Pac23 replied:
fixed.

yvanzo commented on update.sh:
```diff
@@ -24,22 +24,19 @@ then
+    python_command=python3
   else
-    echo Error: python command not found
+    echo Error: python3 command not found , please install python3 or create the virual env with python3
```
```suggestion
    echo Error: python3 command not found , please install python3 or create a virtual env with python3
```

Pac23 commented:
writing the tests now @yvanzo resolved everything else.

Pac23 commented:
changed the base to py3

yvanzo commented:
It is no more necessary to have a separate branch `py3` since SIR doesn’t depend on `master:HEAD` anymore (t depends on `2.20190107.0` release via PyPI).

yvanzo commented:
@Pac23: Thanks for your work. I rebased this pull request on `master` branch, squashed all of your commits together, and added a couple of basic tests.

Pac23 commented:
> @Pac23: Thanks for your work. I rebased this pull request on `master` branch, squashed all of your commits together, and added a couple of basic tests.

sorry if i missed your messages/reply on irc my bouncer has been broken.

yvanzo commented:
No worry @Pac23! Are you still interested in solving [SOLR-112](https://tickets.metabrainz.org/browse/SOLR-112)?

Pac23 commented:
> No worry @Pac23! Are you still interested in solving [SOLR-112](https://tickets.metabrainz.org/browse/SOLR-112)?

yup,jan 1 as a deadline seems bit difficult to make given the eol,i'll try to get this done asap though.
